okhttp-logging-interceptor/src/main/java/okhttp3/logging/HttpLoggingInterceptor.java

```diff
@@ -270,7 +292,11 @@ public Level getLevel() {
 
         if (contentLength != 0) {
           logger.log("");
-          logger.log(buffer.clone().readString(charset));
+          if (buffer.size() <= getBodyLogMax() || getBodyLogMax() < 0) {
```

Comment:
So I still think you are reading the full buffer on line 241/263, so not sure this change is doing what you want.  Have you tested with your large post?

Reply:
This seems like a good point to reduce Long.MAX_VALUE on the line I mention above, avoid additional delay in this interceptor.